src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java

```diff
@@ -0,0 +1,386 @@
+            given(weightsRef.get()).willReturn(futureQuery);
+            willThrow(ExecutionException.class).given(futureQuery).get();
+
+            weightDao.getAllWeight(owner, petName);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
+    public void shouldReturnAllWeightsBetweenDatesOnDatabaseWhenRequested() throws DatabaseAccessException,
+        ExecutionException, InterruptedException {
+        given(db.collection(anyString())).willReturn(usersRef);
+        given(usersRef.document(anyString())).willReturn(ownerRef);
+        given(ownerRef.collection(anyString())).willReturn(petsRef);
+        given(petsRef.document(anyString())).willReturn(petRef);
+        given(petRef.collection(anyString())).willReturn(weightsRef);
+        given(weightsRef.get()).willReturn(futureQuery);
+        given(futureQuery.get()).willReturn(querySnapshot);
+        given(querySnapshot.getDocuments()).willReturn(weightsDocuments);
+        given(weightsDocuments.iterator()).willReturn(it);
+        given(it.hasNext()).willReturn(true);
+        given(it.hasNext()).willReturn(true);
+        given(it.hasNext()).willReturn(false);
+
+        List<Map<String, Object>> list = weightDao.getAllWeightsBetween(owner, petName, date, date2);
+
+        assertEquals(weightList, list, "Should return a List containing all meals between two dates");
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenGetAllWeightsBetweenFromDatabaseReceivesInterruptedException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(weightsRef);
+            given(weightsRef.get()).willReturn(futureQuery);
+            willThrow(InterruptedException.class).given(futureQuery).get();
+
+            weightDao.getAllWeightsBetween(owner, petName, date, date2);
+        }, INTERRUPTED_EXC_MSG);
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenGetAllWeightsBetweenFromDatabaseReceivesExecutionException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(weightsRef);
+            given(weightsRef.get()).willReturn(futureQuery);
+            willThrow(ExecutionException.class).given(futureQuery).get();
+
+            weightDao.getAllWeightsBetween(owner, petName, date, date2);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 17 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)